SimPEG/dask/potential_fields/magnetics/simulation.py

```diff
@@ -0,0 +1,33 @@
+import dask.array as da
+import dask
+import numpy as np
+from ....potential_fields.magnetics import Simulation3DIntegral as Sim
+from ....utils import sdiag, mkvc
+
+def dask_getJtJdiag(self, m, W=None):
+    """
+        Return the diagonal of JtJ
+    """
+
+    self.model = m
+
+    if W is None:
+        W = np.ones(self.nD)
+    else:
+        W = W.diagonal()
+    if getattr(self, "_gtg_diagonal", None) is not None:
+        return self._gtg_diagonal
+
```

Comment:
This is not working for Spherical formulation as the JtJdiag changes every iteration. The `chiDeriv` needs to be applied as it was done on current `simulation`

Reply:
I got ya, would it be better to add these to the `deleteOnModelUpdate` property of the `simulation` class. Contextually of course if chiDeriv is dependent on the model?

Reply:
So the trick here is to keep the `_gtg_diagonal` and just apply it to the non-linear part (i.e. `chiDeriv`) at each iteration. I think we probably want to avoid doing things in a for-loop over the data after this too...

Reply:
Yeah, I gotcha. I’ll get it later tonight then get these pulled in.